casper/src/test/scala/coop/rchain/casper/genesis/contracts/TestUtil.scala

```diff
@@ -93,4 +101,82 @@ object TestUtil {
       _ <- runtime.reducer.phlo
     } yield ()
 
+  def defaultTerms(
+      genesisPk: PublicKey,
+      vaults: Seq[Vault],
+      revSupply: Long,
+      minimumBond: Long,
+      maximumBond: Long
+  ): Seq[DeployData] =
+    Seq(
+      StandardDeploys.listOps,
+      StandardDeploys.either,
+      StandardDeploys.nonNegativeNumber,
+      StandardDeploys.makeMint,
+      StandardDeploys.PoS,
+      StandardDeploys.authKey,
+      StandardDeploys.revVault,
+      StandardDeploys.testRev(genesisPk, vaults, revSupply)
+    )
+
+  def buildGenesis[F[_]: Concurrent: ContextShift, G[_]: Parallel[F, ?[_]]](
+      implicit scheduler: Scheduler
+  ): F[BlockMessage] = {
+
+    implicit val log: Log.NOPLog[F]     = new Log.NOPLog[F]
+    implicit val metricsEff: Metrics[F] = new metrics.Metrics.MetricsNOP[F]
+
+    val storageDirectory  = Files.createTempDirectory(s"hash-set-casper-test-genesis")
+    val storageSize: Long = 1024L * 1024
+
+    for {
+      runtime        <- Runtime.createWithEmptyCost(storageDirectory, storageSize, StoreType.LMDB)
+      runtimeManager <- RuntimeManager.fromRuntime(runtime)
+      genesis        <- TestUtil.defaultGenesisSetup(runtimeManager)
+      _              <- runtime.close()
+    } yield genesis
+  }
+
+  def defaultGenesisSetup[F[_]: Concurrent](runtimeManager: RuntimeManager[F]): F[BlockMessage] = {
+
+    val (_, genesisPk) = Ed25519.newKeyPair
+
+    val (_, validatorPks) = Seq.fill(4)(Ed25519.newKeyPair).unzip
+    val bonds             = createBonds(validatorPks)
+    val validators        = bonds.map { case (pk, stake) => Validator(pk, stake) }.toSeq
+
+    val vaults = Traverse[List]
+      .traverse(validatorPks.toList)(RevAddress.fromPublicKey)
+      .get
+      .map(Vault(_, 1000L))
+    val minimumBond = 0L
+    val maximumBond = Long.MaxValue
+    val revSupply   = Long.MaxValue
+
+    val bondDeploys =
+      validators.map { validator =>
+        DeployData(
+          deployer = ByteString.copyFrom(validator.pk.bytes),
+          term = validator.code,
+          timestamp = System.currentTimeMillis(),
+          phloLimit = Long.MaxValue
+        )
+      }
+
+    val defaultDeploys =
+      defaultTerms(
+        genesisPk,
+        vaults,
+        revSupply,
+        minimumBond,
+        maximumBond
+      ) ++ bondDeploys
+
+    Genesis.withContracts(
```

Comment:
The growth of this test setup worries me. I'd rather imagine the deploys to be computed inside the Genesis methods, based on arguments like `genesisPubKey` or `vaults`. One consequence of having this duplicated is - to state the obvious - there's nothing that keeps it in sync with how it behaves in prod.